From ea334134ccd3da218d8881beea43b939844424fb Mon Sep 17 00:00:00 2001
From: Dmitri Epshtein <dima@marvell.com>
Date: Wed, 8 Aug 2018 15:19:56 +0300
Subject: [PATCH 1/3] examples/ipsec-secgw: add Marvell config files

Add files ep0-mrvl.cfg and ep1-mrvl.cfg as example.

For transport mode 4 flows defined per port to support RSS

cfg          SIP             DIP        RXQ
ep0     192.168.106.10  192.168.171.10   0
        192.168.106.10  192.168.176.10   2
        192.168.106.10  192.168.180.10   1
        192.168.106.10  192.168.175.10   3

ep1     192.168.116.10  192.168.220.10   2
        192.168.116.10  192.168.224.10   0
        192.168.116.10  192.168.228.10   3
        192.168.116.10  192.168.236.10   1

Change-Id: I2434ef9393bf1a60213ae9366a9d1826e7818d91
Signed-off-by: Dmitri Epshtein <dima@marvell.com>
---
 examples/ipsec-secgw/ep0-mrvl.cfg | 148 ++++++++++++++++++++++++++++++++++++++
 examples/ipsec-secgw/ep1-mrvl.cfg | 148 ++++++++++++++++++++++++++++++++++++++
 2 files changed, 296 insertions(+)
 create mode 100644 examples/ipsec-secgw/ep0-mrvl.cfg
 create mode 100644 examples/ipsec-secgw/ep1-mrvl.cfg

diff --git a/examples/ipsec-secgw/ep0-mrvl.cfg b/examples/ipsec-secgw/ep0-mrvl.cfg
new file mode 100644
index 0000000..be9a155
--- /dev/null
+++ b/examples/ipsec-secgw/ep0-mrvl.cfg
@@ -0,0 +1,148 @@
+###########################################################################
+#   IPSEC-SECGW Endpoint sample configuration
+#
+#   The main purpose of this file is to show how to configure two systems
+#   back-to-back that would forward traffic through an IPsec tunnel. This
+#   file is the Endpoint 0 configuration. To use this configuration file,
+#   add the following command-line option:
+#
+#       -f ./ep0.cfg
+#
+###########################################################################
+
+# ************* SP IPv4 rules **************
+
+# IPv4 tunnel OUT mode
+sp ipv4 out esp protect 3 pri 1 dst 192.168.103.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 6 pri 1 dst 192.168.106.0/24 sport 0:65535 dport 0:65535
+
+# IPv4 transport OUT mode
+sp ipv4 out esp protect 10 pri 1 dst 192.168.175.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 11 pri 1 dst 192.168.176.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 12 pri 1 dst 192.168.171.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 13 pri 1 dst 192.168.180.0/24 sport 0:65535 dport 0:65535
+
+# Null IPv4 tunnel OUT mode
+sp ipv4 out esp protect 15 pri 1 dst 192.168.200.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 16 pri 1 dst 192.168.201.0/24 sport 0:65535 dport 0:65535
+
+# Bypass OUT mode
+sp ipv4 out esp bypass pri 1 dst 192.168.240.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp bypass pri 1 dst 192.168.241.0/24 sport 0:65535 dport 0:65535
+
+# IPv4 tunnel IN mode
+sp ipv4 in esp protect 103 pri 1 dst 192.168.113.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 106 pri 1 dst 192.168.116.0/24 sport 0:65535 dport 0:65535
+
+# IPv4 transport IN mode
+sp ipv4 in esp protect 110 pri 1 dst 192.168.220.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 111 pri 1 dst 192.168.224.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 112 pri 1 dst 192.168.228.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 113 pri 1 dst 192.168.236.0/24 sport 0:65535 dport 0:65535
+
+# Null IPv4 tunnel IN mode
+sp ipv4 in esp protect 115 pri 1 dst 192.168.210.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 116 pri 1 dst 192.168.211.0/24 sport 0:65535 dport 0:65535
+
+# Bypass IN mode
+sp ipv4 in esp bypass pri 1 dst 192.168.245.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp bypass pri 1 dst 192.168.246.0/24 sport 0:65535 dport 0:65535
+
+
+# ******** SA rules ***********
+
+# IPv4 tunnel OUT mode
+sa out 3 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+auth_algo sha1-hmac auth_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+mode ipv4-tunnel src 172.16.1.5 dst 172.16.2.5
+
+sa out 6 cipher_algo aes-128-cbc cipher_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
+a0:a0:a0:a0:a0 auth_algo sha1-hmac auth_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
+a0:a0:a0:a0:a0:a0:a0:a0:a0 mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.6
+
+# IPv4 transport OUT mode
+sa out 10 cipher_algo aes-128-cbc cipher_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
+a1:a1:a1:a1:a1 auth_algo sha1-hmac auth_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
+a1:a1:a1:a1:a1:a1:a1:a1:a1 mode transport
+
+sa out 11 cipher_algo aes-128-cbc cipher_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
+b2:b2:b2:b2:b2 auth_algo sha1-hmac auth_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
+b2:b2:b2:b2:b2:b2:b2:b2:b2 mode transport
+
+sa out 12 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3:c3:c3:c3:c3 mode transport
+
+sa out 13 cipher_algo aes-128-cbc cipher_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4 auth_algo sha1-hmac auth_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4:d4:d4:d4:d4 mode transport
+
+# Null IPv4 tunnel OUT mode
+sa out 15 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.5 dst 172.16.2.5
+sa out 16 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.6
+
+# IPv4 tunnel IN mode
+sa in 103 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+auth_algo sha1-hmac auth_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+mode ipv4-tunnel src 172.16.2.5 dst 172.16.1.5
+
+sa in 106 cipher_algo aes-128-cbc cipher_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
+a0:a0:a0:a0:a0 auth_algo sha1-hmac auth_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
+a0:a0:a0:a0:a0:a0:a0:a0:a0 mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.6
+
+# IPv4 transport IN mode
+sa in 110 cipher_algo aes-128-cbc cipher_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
+a1:a1:a1:a1:a1 auth_algo sha1-hmac auth_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
+a1:a1:a1:a1:a1:a1:a1:a1:a1 mode transport
+
+sa in 111 cipher_algo aes-128-cbc cipher_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
+b2:b2:b2:b2:b2 auth_algo sha1-hmac auth_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
+b2:b2:b2:b2:b2:b2:b2:b2:b2 mode transport
+
+sa in 112 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3:c3:c3:c3:c3 mode transport
+
+sa in 113 cipher_algo aes-128-cbc cipher_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4 auth_algo sha1-hmac auth_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4:d4:d4:d4:d4 mode transport
+
+# Null IPv4 tunnel IN mode
+sa in 115 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.5 dst 172.16.1.5
+sa in 116 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.6
+
+
+# ********** Routing rules **************
+
+# IPv4 tunnel OUT mode
+rt ipv4 dst 172.16.2.5/32 port 1
+rt ipv4 dst 172.16.2.6/32 port 1
+
+# IPv4 transport OUT mode
+rt ipv4 dst 192.168.175.0/24 port 1
+rt ipv4 dst 192.168.176.0/24 port 1
+rt ipv4 dst 192.168.171.0/24 port 1
+rt ipv4 dst 192.168.180.0/24 port 1
+
+# Bypass OUT mode
+rt ipv4 dst 192.168.240.0/24 port 1
+rt ipv4 dst 192.168.241.0/24 port 1
+
+# IPv4 tunnel IN mode
+rt ipv4 dst 192.168.113.0/24 port 0
+rt ipv4 dst 192.168.116.0/24 port 0
+
+# IPv4 transport IN mode
+rt ipv4 dst 192.168.220.0/24 port 0
+rt ipv4 dst 192.168.224.0/24 port 0
+rt ipv4 dst 192.168.228.0/24 port 0
+rt ipv4 dst 192.168.236.0/24 port 0
+
+# Null IPv4 tunnel IN mode
+rt ipv4 dst 192.168.210.0/24 port 0
+rt ipv4 dst 192.168.211.0/24 port 0
+
+# Bypass IN mode
+rt ipv4 dst 192.168.245.0/24 port 0
+rt ipv4 dst 192.168.246.0/24 port 0
+
diff --git a/examples/ipsec-secgw/ep1-mrvl.cfg b/examples/ipsec-secgw/ep1-mrvl.cfg
new file mode 100644
index 0000000..f0f7690
--- /dev/null
+++ b/examples/ipsec-secgw/ep1-mrvl.cfg
@@ -0,0 +1,148 @@
+###########################################################################
+#   IPSEC-SECGW Endpoint1 sample configuration
+#
+#   The main purpose of this file is to show how to configure two systems
+#   back-to-back that would forward traffic through an IPsec tunnel. This
+#   file is the Endpoint1 configuration. To use this configuration file,
+#   add the following command-line option:
+#
+#       -f ./ep1.cfg
+#
+###########################################################################
+
+# *********** SP IPv4 rules ***************
+
+# IPv4 tunnel IN mode
+sp ipv4 in esp protect 3 pri 1 dst 192.168.103.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 6 pri 1 dst 192.168.106.0/24 sport 0:65535 dport 0:65535
+
+# IPv4 transport IN mode
+sp ipv4 in esp protect 10 pri 1 dst 192.168.175.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 11 pri 1 dst 192.168.176.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 12 pri 1 dst 192.168.171.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 13 pri 1 dst 192.168.180.0/24 sport 0:65535 dport 0:65535
+
+# Null IPv4 tunnel IN mode
+sp ipv4 in esp protect 15 pri 1 dst 192.168.200.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp protect 16 pri 1 dst 192.168.201.0/24 sport 0:65535 dport 0:65535
+
+# Bypass IN mode
+sp ipv4 in esp bypass dst 192.168.240.0/24 sport 0:65535 dport 0:65535
+sp ipv4 in esp bypass dst 192.168.241.0/24 sport 0:65535 dport 0:65535
+
+# IPv4 tunnel OUT mode
+sp ipv4 out esp protect 103 pri 1 dst 192.168.113.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 106 pri 1 dst 192.168.116.0/24 sport 0:65535 dport 0:65535
+
+# IPv4 transport OUT mode
+sp ipv4 out esp protect 110 pri 1 dst 192.168.220.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 111 pri 1 dst 192.168.224.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 112 pri 1 dst 192.168.228.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 113 pri 1 dst 192.168.236.0/24 sport 0:65535 dport 0:65535
+
+# Null IPv4 tunnel OUT mode
+sp ipv4 out esp protect 115 pri 1 dst 192.168.210.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp protect 116 pri 1 dst 192.168.211.0/24 sport 0:65535 dport 0:65535
+
+# Bypass OUT mode
+sp ipv4 out esp bypass pri 1 dst 192.168.245.0/24 sport 0:65535 dport 0:65535
+sp ipv4 out esp bypass pri 1 dst 192.168.246.0/24 sport 0:65535 dport 0:65535
+
+
+# ************* SA rules ****************
+
+# IPv4 tunnel IN mode
+sa in 3 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+auth_algo sha1-hmac auth_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+mode ipv4-tunnel src 172.16.1.5 dst 172.16.2.5
+
+sa in 6 cipher_algo aes-128-cbc cipher_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
+a0:a0:a0:a0:a0 auth_algo sha1-hmac auth_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
+a0:a0:a0:a0:a0:a0:a0:a0:a0 mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.6
+
+# IPv4 transport IN mode
+sa in 10 cipher_algo aes-128-cbc cipher_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
+a1:a1:a1:a1:a1 auth_algo sha1-hmac auth_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
+a1:a1:a1:a1:a1:a1:a1:a1:a1 mode transport
+
+sa in 11 cipher_algo aes-128-cbc cipher_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
+b2:b2:b2:b2:b2 auth_algo sha1-hmac auth_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
+b2:b2:b2:b2:b2:b2:b2:b2:b2 mode transport
+
+sa in 12 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3:c3:c3:c3:c3 mode transport
+
+sa in 13 cipher_algo aes-128-cbc cipher_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4 auth_algo sha1-hmac auth_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4:d4:d4:d4:d4 mode transport
+
+# Null IPv4 tunnel IN mode
+sa in 15 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.5 dst 172.16.2.5
+sa in 16 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.1.6 dst 172.16.2.6
+
+# IPv4 tunnel OUT mode
+sa out 103 cipher_algo aes-128-cbc cipher_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+auth_algo sha1-hmac auth_key 0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0 \
+mode ipv4-tunnel src 172.16.2.5 dst 172.16.1.5
+
+sa out 106 cipher_algo aes-128-cbc cipher_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
+a0:a0:a0:a0:a0 auth_algo sha1-hmac auth_key a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:a0:\
+a0:a0:a0:a0:a0:a0:a0:a0:a0 mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.6
+
+# IPv4 transport OUT mode
+sa out 110 cipher_algo aes-128-cbc cipher_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
+a1:a1:a1:a1:a1 auth_algo sha1-hmac auth_key a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:a1:\
+a1:a1:a1:a1:a1:a1:a1:a1:a1 mode transport
+
+sa out 111 cipher_algo aes-128-cbc cipher_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
+b2:b2:b2:b2:b2 auth_algo sha1-hmac auth_key b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:b2:\
+b2:b2:b2:b2:b2:b2:b2:b2:b2 mode transport
+
+sa out 112 cipher_algo aes-128-cbc cipher_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3 auth_algo sha1-hmac auth_key c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:c3:\
+c3:c3:c3:c3:c3:c3:c3:c3:c3 mode transport
+
+sa out 113 cipher_algo aes-128-cbc cipher_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4 auth_algo sha1-hmac auth_key d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:d4:\
+d4:d4:d4:d4:d4:d4:d4:d4:d4 mode transport
+
+# Null IPv4 tunnel OUT mode
+sa out 115 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.5 dst 172.16.1.5
+sa out 116 cipher_algo null auth_algo null mode ipv4-tunnel src 172.16.2.6 dst 172.16.1.6
+
+
+# ********* Routing rules **********
+
+# IPv4 tunnel OUT mode
+rt ipv4 dst 172.16.1.5/32 port 1
+rt ipv4 dst 172.16.1.6/32 port 1
+
+# IPv4 transport OUT mode
+rt ipv4 dst 192.168.220.0/24 port 1
+rt ipv4 dst 192.168.224.0/24 port 1
+rt ipv4 dst 192.168.228.0/24 port 1
+rt ipv4 dst 192.168.236.0/24 port 1
+
+# Bypass OUT mode
+rt ipv4 dst 192.168.245.0/24 port 1
+rt ipv4 dst 192.168.246.0/24 port 1
+
+# IPv4 tunnel IN mode
+rt ipv4 dst 192.168.103.0/24 port 0
+rt ipv4 dst 192.168.106.0/24 port 0
+
+# IPv4 transport IN mode
+rt ipv4 dst 192.168.175.0/24 port 0
+rt ipv4 dst 192.168.176.0/24 port 0
+rt ipv4 dst 192.168.171.0/24 port 0
+rt ipv4 dst 192.168.180.0/24 port 0
+
+# Null IPv4 tunnel IN mode
+rt ipv4 dst 192.168.200.0/24 port 0
+rt ipv4 dst 192.168.201.0/24 port 0
+
+# Bypass IN mode
+rt ipv4 dst 192.168.240.0/24 port 0
+rt ipv4 dst 192.168.241.0/24 port 0
+
-- 
2.7.4

